src/main/java/org/cactoos/matchers/TextHasString.java

```diff
@@ -59,17 +69,26 @@ public TextHasString(final String text) {
     public TextHasString(final Matcher<String> mtr) {
         super();
         this.matcher = mtr;
+        this.result = "";
     }
 
     @Override
     public boolean matchesSafely(final Text item) {
-        return this.matcher.matches(new UncheckedText(item).asString());
+        this.result = new UncheckedText(item).asString();
+        return this.matcher.matches(this.result);
     }
 
     @Override
     public void describeTo(final Description description) {
-        description.appendText("Text with ");
+        description.appendText(TextHasString.PREFIX);
         description.appendDescriptionOf(this.matcher);
     }
 
+    @Override
+    public void describeMismatchSafely(
+        final Text item,
```

Comment:
@nick318 What is this method, why have you implemented it? If it is necessary please, write a unit test for it.

Reply:
@amihaiemil The method provide a description for failed case. Before the implementation in failed case there was `toString` representation of Object, as you may assume it was not clear description. 
I added a unit test as you asked.